code/ATMOSPHERICS/atmospherics.dm

```diff
@@ -64,8 +64,8 @@ GLOBAL_DATUM_INIT(pipe_icon_manager, /datum/pipe_icon_manager, new())
 
 // Icons/overlays/underlays
 /obj/machinery/atmospherics/update_icon()
-	var/turf/T = loc
-	if(level == 2 || !T.intact)
+	var/turf/T = get_turf(loc)
+	if(!T || level == 2 || !T.intact)

```

Comment:
Similar to the other statement, why is `update_icon` being called when it's `loc` is something other than a turf?

Reply:
Happens due to blob_act and other reasons destroying the object. The destroy calls disconnect which calls update_icon.